modules/ossm-security-auth-policy.adoc

```diff
@@ -0,0 +1,130 @@
+////
+Module included in the following assemblies:
+-service_mesh/v2x/ossm-security.adoc
+////
+
+[id="ossm-vs-istio_{context}"]
+= Role Based Access Control (RBAC)
+
+Role-based access control (RBAC) objects determine whether a user or service is allowed to perform a given action within a project. You can define mesh-, namespace-, and workload-wide access control for your workloads in the mesh. 
+
+To configure RBAC, create an `AuthorizationPolicy` custom resource in the namespace for which you are configuring access. If you are configuring mesh-wide access, use `istio-system`. The namespace must match your `metadata.namespace` field.
+
+An authorization policy includes a selector, an action, and a list of rules:
+
+* The `selector` field specifies the target of the policy.
+* The `action` field specifies whether to allow or deny the request.
+* The `rules` specify when to trigger the action.
+** The `from` field in the rules specifies the sources of the request.
+** The `to` field in the rules specifies the operations of the request.
+** The `when` field specifies the conditions needed to apply the rule.
+
+== Configure intra-project communication
+
+You can use authorization policies to configure your control plane to allow or deny the traffic communicating with your mesh or services in your mesh. 
+
+=== Restrict access to services outside a namespace
+
+You can deny requests from any source that is not in the `bookinfo` namespace with the following `AuthorizationPolicy` custom resource example.
+
+[source,yaml]
+----
+apiVersion: security.istio.io/v1beta1
+kind: AuthorizationPolicy
+metadata:
+ name: httpbin-deny
+ namespace: bookinfo
+spec:
+ selector:
+   matchLabels:
+     app: httpbin
+     version: v1
+ action: DENY
+ rules:
+ - from:
+   - source:
+       notNamespaces: ["bookinfo"]
+----
+
+=== Creating allow-all and default deny-all authorization policies
+
+The following example shows an allow-all authorization policy that allows full access to all workloads in the default namespace.
+
+[source,yaml]
+----
+apiVersion: security.istio.io/v1beta1
+kind: AuthorizationPolicy
+metadata:
+  name: allow-all
+  namespace: bookinfo
+spec:
+  action: ALLOW
+  rules:
+  - {}
+----
+
+The following example shows a policy that doesn’t allow any access to all workloads in the `bookinfo` namespace.
+
+[source,yaml]
+----
+apiVersion: security.istio.io/v1beta1
+kind: AuthorizationPolicy
+metadata:
+  name: deny-all
+  namespace: bookinfo
+spec:
+  {}
+----
+
+== Create a role for the ingress gateway
+
+== Set restricted access with JWT
+
+[source,yaml]
+----
+apiVersion: "security.istio.io/v1beta1"
+kind: "RequestAuthentication"
```

Comment:
We should probably explain `RequestAuthentication` here as well.

there's a trick to do this IIRC... the `RequestAuthentication` specifies a rule for the principal (ie the token). but if you simply don't send a token, the request will go through, so you'll need an AuthorizationPolicy as well to make sure that the user did send a token (require a principal). see here for an example: https://github.com/dgn/oidc-filter/blob/master/example/istio-auth.yaml

Reply:
@dgn Can you point me to an explanation of `RequestAuthentication`?